store-implementation/map-store/src/main/java/uk/gov/gchq/gaffer/mapstore/impl/MapImpl.java

```diff
@@ -15,61 +15,248 @@
  */
 package uk.gov.gchq.gaffer.mapstore.impl;
 
+import com.google.common.collect.Sets;
 import uk.gov.gchq.gaffer.data.element.Element;
-import uk.gov.gchq.gaffer.data.element.Properties;
+import uk.gov.gchq.gaffer.data.element.GroupedProperties;
 import uk.gov.gchq.gaffer.data.element.id.EdgeId;
 import uk.gov.gchq.gaffer.data.element.id.EntityId;
 import uk.gov.gchq.gaffer.mapstore.MapStoreProperties;
-import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.mapstore.factory.MapFactory;
+import uk.gov.gchq.gaffer.mapstore.factory.SimpleMapFactory;
+import uk.gov.gchq.gaffer.mapstore.multimap.MultiMap;
 import uk.gov.gchq.gaffer.store.schema.Schema;
 import uk.gov.gchq.gaffer.store.schema.SchemaElementDefinition;
+import uk.gov.gchq.gaffer.store.util.AggregatorUtil;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Stream;
 
 /**
  * The internal variables of this class are package-private. This allows operation handlers for the
  * {@link uk.gov.gchq.gaffer.mapstore.MapStore} to be placed in the same package and get access to the maps, without
  * exposing the internal state of the MapStore to classes outside of this package.
  */
 public class MapImpl {
+    public static final String AGG_ELEMENTS = "aggElements";
+    public static final String NON_AGG_ELEMENTS = "nonAggElements";
+    public static final String ENTITY_ID_TO_ELEMENTS = "entityIdToElements";
+    public static final String EDGE_ID_TO_ELEMENTS = "edgeIdToElements";
+    public static final Set<String> MAP_NAMES = Collections.unmodifiableSet(
+            Sets.newHashSet(AGG_ELEMENTS, NON_AGG_ELEMENTS));
+    public static final Set<String> MULTI_MAP_NAMES = Collections.unmodifiableSet(
+            Sets.newHashSet(ENTITY_ID_TO_ELEMENTS, EDGE_ID_TO_ELEMENTS));
```

Comment:
I added these in when I was investigating MapDB and Hazelcast implementations, probably don't really need them.